lib/ravelin/ravelin_object.rb

```diff
@@ -0,0 +1,50 @@
+module Ravelin
+  class RavelinObject
+    class << self
+      attr_reader :attributes
+
+      def required_attributes
+        @required_attributes ||= []
+      end
+
+      def attr_required(*names)
+        @required_attributes = *names
+      end
+
+      def attr_accessor(*names)
```

Comment:
Cool, I was a little worried about the instance variables defined elsewhere polluting the `serialised_hash`. 

Structs and members looks good! I'll keep that in mind for next time or if there is ever a refactor. 

For now I'll go with option 1 (as it is) and make the variable name a bit more fitting and less likely to conflict elsewhere. `attributes` isn't the best name.